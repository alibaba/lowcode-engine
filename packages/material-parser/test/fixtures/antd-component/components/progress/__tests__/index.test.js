import React from 'react';
import { mount } from 'enzyme';
import Progress from '..';
import { handleGradient, sortGradient } from '../Line';
import mountTest from '../../../tests/shared/mountTest';
import rtlTest from '../../../tests/shared/rtlTest';

describe('Progress', () => {
  mountTest(Progress);
  rtlTest(Progress);

  it('successPercent should decide the progress status when it exists', () => {
    const wrapper = mount(<Progress percent={100} successPercent={50} />);
    expect(wrapper.find('.ant-progress-status-success')).toHaveLength(0);

    wrapper.setProps({ percent: 50, successPercent: 100 });
    expect(wrapper.find('.ant-progress-status-success')).toHaveLength(1);

    wrapper.setProps({ percent: 100, successPercent: 0 });
    expect(wrapper.find('.ant-progress-status-success')).toHaveLength(0);
  });

  it('render out-of-range progress', () => {
    const wrapper = mount(<Progress percent={120} />);
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('render out-of-range progress with info', () => {
    const wrapper = mount(<Progress percent={120} showInfo />);
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('render negative progress', () => {
    const wrapper = mount(<Progress percent={-20} />);
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('render negative successPercent', () => {
    const wrapper = mount(<Progress percent={50} successPercent={-20} />);
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('render format', () => {
    const wrapper = mount(
      <Progress
        percent={50}
        successPercent={10}
        format={(percent, successPercent) => `${percent} ${successPercent}`}
      />,
    );
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('render strokeColor', () => {
    const wrapper = mount(<Progress type="circle" percent={50} strokeColor="red" />);
    expect(wrapper.render()).toMatchSnapshot();
    wrapper.setProps({
      strokeColor: {
        from: '#108ee9',
        to: '#87d068',
      },
      type: 'line',
    });
    expect(wrapper).toMatchSnapshot();
    wrapper.setProps({
      strokeColor: {
        '0%': '#108ee9',
        '100%': '#87d068',
      },
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('render normal progress', () => {
    const wrapper = mount(<Progress status="normal" />);
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('render trailColor progress', () => {
    const wrapper = mount(<Progress status="normal" trailColor="#ffffff" />);
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('render dashboard zero gapDegree', () => {
    const wrapper = mount(<Progress type="dashboard" gapDegree={0} />);
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('render dashboard 295 gapDegree', () => {
    const wrapper = mount(<Progress type="dashboard" gapDegree={295} />);
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('render dashboard 296 gapDegree', () => {
    const wrapper = mount(<Progress type="dashboard" gapDegree={296} />);
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('get correct line-gradient', () => {
    expect(handleGradient({ from: 'test', to: 'test' }).backgroundImage).toBe(
      'linear-gradient(to right, test, test)',
    );
    expect(handleGradient({}).backgroundImage).toBe('linear-gradient(to right, #1890ff, #1890ff)');
    expect(handleGradient({ from: 'test', to: 'test', '0%': 'test' }).backgroundImage).toBe(
      'linear-gradient(to right, test 0%)',
    );
  });

  it('sort gradients correctly', () => {
    expect(sortGradient({ '10%': 'test10', '30%': 'test30', '20%': 'test20' })).toBe(
      'test10 10%, test20 20%, test30 30%',
    );
    expect(sortGradient({ '10%': 'test10', '30%': 'test30', '20%': 'test20', dummy: 'test' })).toBe(
      'test10 10%, test20 20%, test30 30%',
    );
  });

  it('should show success status when percent is 100', () => {
    const wrapper = mount(<Progress percent={100} />);
    expect(wrapper.find('.ant-progress-status-success')).toHaveLength(1);
  });

  // https://github.com/ant-design/ant-design/issues/15950
  it('should show success status when percent is 100 and status is undefined', () => {
    const wrapper = mount(<Progress percent={100} status={undefined} />);
    expect(wrapper.find('.ant-progress-status-success')).toHaveLength(1);
  });

  // https://github.com/ant-design/ant-design/pull/15951#discussion_r273062969
  it('should show success status when status is invalid', () => {
    const errorSpy = jest.spyOn(console, 'error').mockImplementation(() => {});
    const wrapper = mount(<Progress percent={100} status="invalid" />);
    expect(wrapper.find('.ant-progress-status-success')).toHaveLength(1);
    errorSpy.mockRestore();
  });
});
