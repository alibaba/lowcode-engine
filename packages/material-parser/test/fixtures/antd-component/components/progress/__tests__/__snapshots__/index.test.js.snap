// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress render dashboard 295 gapDegree 1`] = `
<div
  class="ant-progress ant-progress-circle ant-progress-status-normal ant-progress-show-info ant-progress-default"
>
  <div
    class="ant-progress-inner"
    style="width: 120px; height: 120px; font-size: 24px;"
  >
    <svg
      class="ant-progress-circle "
      viewBox="0 0 100 100"
    >
      <path
        class="ant-progress-circle-trail"
        d="M 50,50 m 0,47
   a 47,47 0 1 1 0,-94
   a 47,47 0 1 1 0,94"
        fill-opacity="0"
        stroke-linecap="round"
        stroke-width="6"
        style="stroke-dasharray: 0.3097094374405742px 295.3097094374406px; stroke-dashoffset: -147.5px; transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s;"
      />
      <path
        class="ant-progress-circle-path"
        d="M 50,50 m 0,47
   a 47,47 0 1 1 0,-94
   a 47,47 0 1 1 0,94"
        fill-opacity="0"
        stroke=""
        stroke-linecap="round"
        stroke-width="0"
        style="stroke-dasharray: 0px 295.3097094374406px; stroke-dashoffset: -147.5px; transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s;"
      />
    </svg>
    <span
      class="ant-progress-text"
      title="0%"
    >
      0%
    </span>
  </div>
</div>
`;

exports[`Progress render dashboard 296 gapDegree 1`] = `
<div
  class="ant-progress ant-progress-circle ant-progress-status-normal ant-progress-show-info ant-progress-default"
>
  <div
    class="ant-progress-inner"
    style="width: 120px; height: 120px; font-size: 24px;"
  >
    <svg
      class="ant-progress-circle "
      viewBox="0 0 100 100"
    >
      <path
        class="ant-progress-circle-trail"
        d="M 50,50 m 0,47
   a 47,47 0 1 1 0,-94
   a 47,47 0 1 1 0,94"
        fill-opacity="0"
        stroke-linecap="round"
        stroke-width="6"
        style="stroke-dasharray: -0.6902905625594258px 295.3097094374406px; stroke-dashoffset: -148px; transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s;"
      />
      <path
        class="ant-progress-circle-path"
        d="M 50,50 m 0,47
   a 47,47 0 1 1 0,-94
   a 47,47 0 1 1 0,94"
        fill-opacity="0"
        stroke=""
        stroke-linecap="round"
        stroke-width="0"
        style="stroke-dasharray: 0px 295.3097094374406px; stroke-dashoffset: -148px; transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s;"
      />
    </svg>
    <span
      class="ant-progress-text"
      title="0%"
    >
      0%
    </span>
  </div>
</div>
`;

exports[`Progress render dashboard zero gapDegree 1`] = `
<div
  class="ant-progress ant-progress-circle ant-progress-status-normal ant-progress-show-info ant-progress-default"
>
  <div
    class="ant-progress-inner"
    style="width: 120px; height: 120px; font-size: 24px;"
  >
    <svg
      class="ant-progress-circle "
      viewBox="0 0 100 100"
    >
      <path
        class="ant-progress-circle-trail"
        d="M 50,50 m 0,47
   a 47,47 0 1 1 0,-94
   a 47,47 0 1 1 0,94"
        fill-opacity="0"
        stroke-linecap="round"
        stroke-width="6"
        style="stroke-dasharray: 295.3097094374406px 295.3097094374406px; stroke-dashoffset: -0px; transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s;"
      />
      <path
        class="ant-progress-circle-path"
        d="M 50,50 m 0,47
   a 47,47 0 1 1 0,-94
   a 47,47 0 1 1 0,94"
        fill-opacity="0"
        stroke=""
        stroke-linecap="round"
        stroke-width="0"
        style="stroke-dasharray: 0px 295.3097094374406px; stroke-dashoffset: -0px; transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s;"
      />
    </svg>
    <span
      class="ant-progress-text"
      title="0%"
    >
      0%
    </span>
  </div>
</div>
`;

exports[`Progress render format 1`] = `
<div
  class="ant-progress ant-progress-line ant-progress-status-normal ant-progress-show-info ant-progress-default"
>
  <div
    class="ant-progress-outer"
  >
    <div
      class="ant-progress-inner"
    >
      <div
        class="ant-progress-bg"
        style="width: 50%; height: 8px;"
      />
      <div
        class="ant-progress-success-bg"
        style="width: 10%; height: 8px;"
      />
    </div>
  </div>
  <span
    class="ant-progress-text"
    title="50 10"
  >
    50 10
  </span>
</div>
`;

exports[`Progress render negative progress 1`] = `
<div
  class="ant-progress ant-progress-line ant-progress-status-normal ant-progress-show-info ant-progress-default"
>
  <div
    class="ant-progress-outer"
  >
    <div
      class="ant-progress-inner"
    >
      <div
        class="ant-progress-bg"
        style="width: 0%; height: 8px;"
      />
    </div>
  </div>
  <span
    class="ant-progress-text"
    title="0%"
  >
    0%
  </span>
</div>
`;

exports[`Progress render negative successPercent 1`] = `
<div
  class="ant-progress ant-progress-line ant-progress-status-normal ant-progress-show-info ant-progress-default"
>
  <div
    class="ant-progress-outer"
  >
    <div
      class="ant-progress-inner"
    >
      <div
        class="ant-progress-bg"
        style="width: 50%; height: 8px;"
      />
      <div
        class="ant-progress-success-bg"
        style="width: 0%; height: 8px;"
      />
    </div>
  </div>
  <span
    class="ant-progress-text"
    title="50%"
  >
    50%
  </span>
</div>
`;

exports[`Progress render normal progress 1`] = `
<div
  class="ant-progress ant-progress-line ant-progress-status-normal ant-progress-show-info ant-progress-default"
>
  <div
    class="ant-progress-outer"
  >
    <div
      class="ant-progress-inner"
    >
      <div
        class="ant-progress-bg"
        style="width: 0%; height: 8px;"
      />
    </div>
  </div>
  <span
    class="ant-progress-text"
    title="0%"
  >
    0%
  </span>
</div>
`;

exports[`Progress render out-of-range progress 1`] = `
<div
  class="ant-progress ant-progress-line ant-progress-status-success ant-progress-show-info ant-progress-default"
>
  <div
    class="ant-progress-outer"
  >
    <div
      class="ant-progress-inner"
    >
      <div
        class="ant-progress-bg"
        style="width: 100%; height: 8px;"
      />
    </div>
  </div>
  <span
    class="ant-progress-text"
  >
    <span
      aria-label="check-circle"
      class="anticon anticon-check-circle"
      role="img"
    >
      <svg
        aria-hidden="true"
        class=""
        data-icon="check-circle"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm193.5 301.7l-210.6 292a31.8 31.8 0 01-51.7 0L318.5 484.9c-3.8-5.3 0-12.7 6.5-12.7h46.9c10.2 0 19.9 4.9 25.9 13.3l71.2 98.8 157.2-218c6-8.3 15.6-13.3 25.9-13.3H699c6.5 0 10.3 7.4 6.5 12.7z"
        />
      </svg>
    </span>
  </span>
</div>
`;

exports[`Progress render out-of-range progress with info 1`] = `
<div
  class="ant-progress ant-progress-line ant-progress-status-success ant-progress-show-info ant-progress-default"
>
  <div
    class="ant-progress-outer"
  >
    <div
      class="ant-progress-inner"
    >
      <div
        class="ant-progress-bg"
        style="width: 100%; height: 8px;"
      />
    </div>
  </div>
  <span
    class="ant-progress-text"
  >
    <span
      aria-label="check-circle"
      class="anticon anticon-check-circle"
      role="img"
    >
      <svg
        aria-hidden="true"
        class=""
        data-icon="check-circle"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm193.5 301.7l-210.6 292a31.8 31.8 0 01-51.7 0L318.5 484.9c-3.8-5.3 0-12.7 6.5-12.7h46.9c10.2 0 19.9 4.9 25.9 13.3l71.2 98.8 157.2-218c6-8.3 15.6-13.3 25.9-13.3H699c6.5 0 10.3 7.4 6.5 12.7z"
        />
      </svg>
    </span>
  </span>
</div>
`;

exports[`Progress render strokeColor 1`] = `
<div
  class="ant-progress ant-progress-circle ant-progress-status-normal ant-progress-show-info ant-progress-default"
>
  <div
    class="ant-progress-inner"
    style="width: 120px; height: 120px; font-size: 24px;"
  >
    <svg
      class="ant-progress-circle "
      viewBox="0 0 100 100"
    >
      <path
        class="ant-progress-circle-trail"
        d="M 50,50 m 0,-47
   a 47,47 0 1 1 0,94
   a 47,47 0 1 1 0,-94"
        fill-opacity="0"
        stroke-linecap="round"
        stroke-width="6"
        style="stroke-dasharray: 295.3097094374406px 295.3097094374406px; stroke-dashoffset: -0px; transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s;"
      />
      <path
        class="ant-progress-circle-path"
        d="M 50,50 m 0,-47
   a 47,47 0 1 1 0,94
   a 47,47 0 1 1 0,-94"
        fill-opacity="0"
        stroke=""
        stroke-linecap="round"
        stroke-width="6"
        style="stroke: red; stroke-dasharray: 147.6548547187203px 295.3097094374406px; stroke-dashoffset: -0px; transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s;"
      />
    </svg>
    <span
      class="ant-progress-text"
      title="50%"
    >
      50%
    </span>
  </div>
</div>
`;

exports[`Progress render strokeColor 2`] = `
<Progress
  percent={50}
  showInfo={true}
  size="default"
  strokeColor={
    Object {
      "from": "#108ee9",
      "to": "#87d068",
    }
  }
  strokeLinecap="round"
  trailColor={null}
  type="line"
>
  <div
    className="ant-progress ant-progress-line ant-progress-status-normal ant-progress-show-info ant-progress-default"
  >
    <Line
      percent={50}
      prefixCls="ant-progress"
      showInfo={true}
      size="default"
      strokeColor={
        Object {
          "from": "#108ee9",
          "to": "#87d068",
        }
      }
      strokeLinecap="round"
      trailColor={null}
      type="line"
    >
      <div
        className="ant-progress-outer"
      >
        <div
          className="ant-progress-inner"
        >
          <div
            className="ant-progress-bg"
            style={
              Object {
                "backgroundImage": "linear-gradient(to right, #108ee9, #87d068)",
                "borderRadius": "",
                "height": 8,
                "width": "50%",
              }
            }
          />
        </div>
      </div>
      <span
        className="ant-progress-text"
        title="50%"
      >
        50%
      </span>
    </Line>
  </div>
</Progress>
`;

exports[`Progress render strokeColor 3`] = `
<Progress
  percent={50}
  showInfo={true}
  size="default"
  strokeColor={
    Object {
      "0%": "#108ee9",
      "100%": "#87d068",
    }
  }
  strokeLinecap="round"
  trailColor={null}
  type="line"
>
  <div
    className="ant-progress ant-progress-line ant-progress-status-normal ant-progress-show-info ant-progress-default"
  >
    <Line
      percent={50}
      prefixCls="ant-progress"
      showInfo={true}
      size="default"
      strokeColor={
        Object {
          "0%": "#108ee9",
          "100%": "#87d068",
        }
      }
      strokeLinecap="round"
      trailColor={null}
      type="line"
    >
      <div
        className="ant-progress-outer"
      >
        <div
          className="ant-progress-inner"
        >
          <div
            className="ant-progress-bg"
            style={
              Object {
                "backgroundImage": "linear-gradient(to right, #108ee9 0%, #87d068 100%)",
                "borderRadius": "",
                "height": 8,
                "width": "50%",
              }
            }
          />
        </div>
      </div>
      <span
        className="ant-progress-text"
        title="50%"
      >
        50%
      </span>
    </Line>
  </div>
</Progress>
`;

exports[`Progress render trailColor progress 1`] = `
<div
  class="ant-progress ant-progress-line ant-progress-status-normal ant-progress-show-info ant-progress-default"
>
  <div
    class="ant-progress-outer"
  >
    <div
      class="ant-progress-inner"
      style="background-color: rgb(255, 255, 255);"
    >
      <div
        class="ant-progress-bg"
        style="width: 0%; height: 8px;"
      />
    </div>
  </div>
  <span
    class="ant-progress-text"
    title="0%"
  >
    0%
  </span>
</div>
`;

exports[`Progress rtl render component should be rendered correctly in RTL direction 1`] = `
<div
  class="ant-progress ant-progress-line ant-progress-status-normal ant-progress-show-info ant-progress-default ant-progress-rtl"
>
  <div
    class="ant-progress-outer"
  >
    <div
      class="ant-progress-inner"
    >
      <div
        class="ant-progress-bg"
        style="width:0%;height:8px;border-radius:"
      />
    </div>
  </div>
  <span
    class="ant-progress-text"
    title="0%"
  >
    0%
  </span>
</div>
`;
